### PR TITLE
Bumping versions for Kotlin 1.6.10

### PR DESCRIPTION
**Pull request by MarcoSignoretto**
Bumping versions:

- Kotlin to 1.6.10
- AtomicFu to 0.17.0
- KotlinPoet to 1.10.2
- Kotlin metadata to 0.2.0

### CODE REVIEW

**Comment by minaEweida** [comment]
why not upgrade gradle as well

**Review by minaEweida — APPROVED** [review]

File: build.gradle.kts [hunk 1]
```diff
@@ -37,6 +37,8 @@ allprojects {
     }
 
     tasks.withType<org.jetbrains.kotlin.gradle.tasks.KotlinCompile>().configureEach {
-        kotlinOptions.allWarningsAsErrors = true
```
Comment (nikolaykasyanov) [inline build.gradle.kts:40]: No longer works?

File: mockingbird-compiler/build.gradle.kts [hunk 2]
```diff
@@ -38,7 +38,12 @@ repositories {
 }
 
 tasks.withType<org.jetbrains.kotlin.gradle.tasks.KotlinCompile>().configureEach {
-    kotlinOptions.freeCompilerArgs += "-Xopt-in=kotlin.RequiresOptIn"
+    kotlinOptions {
+        freeCompilerArgs = freeCompilerArgs + "-Xopt-in=kotlin.RequiresOptIn"
+        freeCompilerArgs = freeCompilerArgs + "-language-version=${libs.versions.kotlinTarget.get()}"
+        freeCompilerArgs = freeCompilerArgs + "-api-version=${libs.versions.kotlinTarget.get()}"
```
Comment (nikolaykasyanov) [inline mockingbird-compiler/build.gradle.kts:44]: Nit, how about

```
freeComilerArgs = freeCompilerArgs + listOf(
    ...,
    ...,
    ...,
)```

**Review by nikolaykasyanov — APPROVED** [review]

File: build.gradle.kts (hunk 1 above)
Comment (nikolaykasyanov) [reply build.gradle.kts:40]: or just for consistency?

Comment (MarcoSignoretto) [reply build.gradle.kts:40]: it works on both ways I just prefer this syntax

**Pull request merged**